content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -382,33 +382,33 @@ An important remark about partial SRG blocks is that when the compiler, **AZSLc*
         OUT.m_color = PassSrg::GetColor();
         return OUT;
     }
-  
-## <a name="rootconstant"></a>Root Constants
+```
+## Root Constants
 AZSL supports the definition of shader constants that conform to the DX12 concept of Root Constants: https://docs.microsoft.com/en-us/windows/win32/direct3d12/using-constants-directly-in-the-root-signature.  
   
 Root Constants, declared with the `rootconstant` keyword, will be transparently embedded into a `ConstantBuffer`. They are useful to define shader constants of frequent access. Because they are located in the root signature they can be read instantly without requiring additional levels of indirection.  
   
 Shader constants qualified as `rootconstant` are always declared globally. Examples:  
-  
+```cpp
     rootconstant float4x4 s_objectMatrix;
     rootconstant uint s_materialIndex;
-  
+```
 Root Constants should be used sparingly because the space to define root constants, the Root Signature, is shared with Root Descriptors and Descriptor Tables. In D3D12, the Root Signature is limited at 64 DWords (256 Bytes) worth of space that is shared among Root Constants, Root Descriptors and Descriptor Tables.  
   
-## <a name="option"></a>Shader Variant Options
+## <a name="option"></a>Shader variant options
 
-Before describing how to use Shader Variant Options in a shader, you should read this: [AZSL: Design Principles And Purpose Of Shader Variant Options.](shader-variant-options)  
+Before describing how to use Shader variant options in a shader, you should read this: [AZSL: Design Principles And Purpose Of Shader variant options.](shader-variant-options)  
   
-Shader Variant Options are declared as global variables with the `option` keyword. Their ultimate purpose is to avoid conditional branching, in favor of static execution, all for the sake of performance.  
+Shader variant options are declared as global variables with the `option` keyword. Their ultimate purpose is to avoid conditional branching, in favor of static execution, all for the sake of performance.  
   
 Although the ultimate purpose is to avoid conditional branching at runtime, they can be used to conditionally branch at runtime and, in fact, it's the default behavior of the shader code to conditionally branch at runtime when a Shader Variant has not been compiled (not *baked*).  
```

Comment:
1. I'm unsure about how to use Shader variant options based on this description. It says that shader variant options are used to avoid conditional branching, but that its default behavior is to conditionally branch. Does that mean we need to "enable" stader variant options somehow? 

    My assumption is that shader constants conditionally branch, and to avoid that we define them with `option` keyword. This defines them as shader variant options, which avoid branching at conditional statements. **Can you verify that the suggestion is correct and make adjustments if necessary?**

2. Add "For more info..." if you decide to remove the first sentence, "Before describing....". 

```suggestion
Shader variant options are shader constants that are optimized for conditional statements. By default, shader constants conditionally branch at runtime. To avoid conditional branching and improve performance, Shader variant options use static execution instead of conditional branching. For more information on shader variant options, refer to [AZSL: Design Principles And Purpose Of Shader variant options.](shader-variant-options).
```

Reply:
Changed it all to this:

Shader variant options are declared as global variables with the `option` keyword. Their ultimate purpose is to be used as statically defined flags. It is helpful to think of Shader variant options as compile time configurable C Macros.  
By default, during shader asset compilation, all shader variant options are compiled as global shader constant flags that can be used for conditional branching. For the sake of performance, You can define the values of each shader variant option at compilation time, and thanks to shader compiler optimizations the conditional branches are removed in favor of static execution of some sections of the code.